src/processes/follow-topic-process.js

```diff
@@ -92,9 +94,49 @@ const sendMessageAsAnonymous = async (serverClient, communityMessageFormat, data
   }
   // }
 };
+
+const sendSignedMessage = async ({
+  serverChannel,
+  targetUserId,
+  targetUsername,
+  communityMessageFormatUserId,
+  text
+}) => {
+  const message = await serverChannel?.sendMessage(
+    {
+      text,
+      user_id: communityMessageFormatUserId,
+      is_auto_message: true
+    },
+    {skip_push: true}
+  );
+
+  const notificationPayload = {
+    title: targetUsername,
+    body: `${text?.substring(0, 100)}`
+  };
+
+  const dataPayload = {
+    type: 'message.new',
+    channel_id: serverChannel?.id,
+    message: text,
+    message_schema: 'text',
+    attachment: '',
+    created_at: message?.message?.created_at || new Date().toISOString(),
+    is_annoymous: 'false',
+    priority: 'high',
+    content_available: 'true'
+  };
+
+  sendChatNotificationByChannelMembers([targetUserId], {
+    notification: notificationPayload,
+    data: dataPayload
+  });
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendSignedMessage` function sends a message and then immediately triggers a notification without checking if the message was successfully sent. This could lead to situations where notifications are sent for messages that failed to send. Consider adding error handling or a success check before sending the notification.
```diff
  const sendSignedMessage = async ({
    serverChannel,
    targetUserId,
    targetUsername,
    communityMessageFormatUserId,
    text
  }) => {
    const message = await serverChannel?.sendMessage(
      {
        text,
        user_id: communityMessageFormatUserId,
        is_auto_message: true
      },
      {skip_push: true}
    );

+   if (!message) {
+     throw new Error('Failed to send message');
+   }

    const notificationPayload = {
      title: targetUsername,
      body: `${text?.substring(0, 100)}`
    };

    const dataPayload = {
      type: 'message.new',
      channel_id: serverChannel?.id,
      message: text,
      message_schema: 'text',
      attachment: '',
      created_at: message?.message?.created_at || new Date().toISOString(),
      is_annoymous: 'false',
      priority: 'high',
      content_available: 'true'
    };

    sendChatNotificationByChannelMembers([targetUserId], {
      notification: notificationPayload,
      data: dataPayload
    });
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->